src/main/java/org/owasp/validator/html/Policy.java

```diff
@@ -735,6 +766,25 @@ public Attribute getGlobalAttributeByName(String name) {
     }
 
     /**
+     * A method for returning one of the dynamic <global-attribute> entries by
+     * name.
+     *
+     * @param name The name of the dynamic global-attribute we want to look up.
+     * @return An Attribute associated with the global-attribute lookup name specified,
+     * or null if not found.
+     */
+    public Attribute getDynamicAttributeByName(String name) {
+        Attribute dynamicAttribute = null;
+        for (String d : dynamicAttributes.keySet()) {
+            if (name.startsWith(d)) {
```

Comment:
Ignore above comment -- I get it now. The PR is really about allowing you do add any sort of tag that has a common prefix, not necessarily just "data-", so the suffixes aren't known beforehand.